### PR TITLE
Add support for additional Unlocked Package install options

### PR DESCRIPTION
**Pull request by davidmreed**
# Critical Changes

# Changes

- We added the `apex_compile_type` and `upgrade_type` options for performing Unlocked Package installations.

# Issues Closed

### CODE REVIEW

File: cumulusci/salesforce_api/tests/test_package_install.py [hunk 1]
```diff
@@ -183,4 +185,5 @@ def test_package_install_options_from_task_options():
         name_conflict_resolution=NameConflictResolution.RENAME,
         password="foo",
         security_type=SecurityType.PUSH,
+        apex_compile_type=ApexCompileType.PACKAGE,
     )
```
Comment (davisagli) [inline cumulusci/salesforce_api/tests/test_package_install.py:189]: Include upgrade_type in the test too?

**Review by davisagli — CHANGES_REQUESTED** [review]

File: cumulusci/salesforce_api/tests/test_package_install.py (hunk 1 above)
Comment (davidmreed) [reply cumulusci/salesforce_api/tests/test_package_install.py:189]: I deliberately left it out to validate that there wasn't any negative behavior with its nullability.

Comment (davidmreed) [reply cumulusci/salesforce_api/tests/test_package_install.py:189]: Perhaps I should make that test explicit.

Comment (davisagli) [reply cumulusci/salesforce_api/tests/test_package_install.py:189]: Ah okay. Wouldn't hurt to have both tests but we don't need to hold this up.

**Review by davisagli — DISMISSED** [review]
@davidmreed approved, but did you want to update version.txt to make a dev release?

**Review by davisagli — APPROVED** [review]

**Pull request merged**